examples/example.c

```diff
@@ -28,6 +28,25 @@
 #    define sleep_s(SECONDS) sleep(SECONDS)
 #endif
 
+static double
+traces_sampler_callback(sentry_transaction_context_t *transaction_ctx,
+    sentry_value_t custom_sampling_ctx, const bool *parent_sampled)
+{
+    (void)transaction_ctx; // unused for now
```

Comment:
Primarily because the getters are not exposed, right? Could we start by exposing getters for `name` and `operation` and forget about the external trace inheritance? This way, we could provide an example that does something side-effect free as a first step like
```suggestion
    if (strcmp(sentry_transaction_context_get_name(tx_ctx), "little.teapot")
        != 0) {
        return 0;
    }
```

Reply:
Yes indeed, I think exposing the getters in this PR makes sense (as there's now a reason for them to be exposed).

Reply:
Added the getters in https://github.com/getsentry/sentry-native/pull/1108/commits/a9d72219f122893d1f128630cbddabd5bf8adc9b and tested them in the context of the traces sampler in https://github.com/getsentry/sentry-native/pull/1108/commits/370b16264e2476e3dada827829e69ce659e7b880